### PR TITLE
add EventSourcedBehaviorTimersSpec, #24687

### PR DESCRIPTION
**Pull request by patriknw**
Refs #24687

### CODE REVIEW

**Review by chbatey — APPROVED** [review]
LGTM

File: akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/EventSourcedBehaviorTimersSpec.scala [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    with WordSpecLike {
+
+  import EventSourcedBehaviorTimersSpec._
+
+  val pidCounter = new AtomicInteger(0)
+  private def nextPid(): PersistenceId = PersistenceId(s"c${pidCounter.incrementAndGet()})")
+
+  import akka.actor.typed.scaladsl.adapter._
+  // needed for the untyped event filter
+  private implicit val untypedSystem: akka.actor.ActorSystem = system.toUntyped
```
Comment (johanandren) [inline akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/EventSourcedBehaviorTimersSpec.scala]: add the issue number we have in the other places now

File: akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/EventSourcedBehaviorTimersSpec.scala [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        ref ! s"cmd-$n"
+      }
+      probe.receiveMessages(20)
+
+      // start new instance that is likely to stash the timer message while replaying
+      spawn(testTimerFromSetupBehavior(pid, probe.ref))
+      probe.expectMessage("scheduled")
+    }
+
+  }
```
Comment (johanandren) [inline akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/EventSourcedBehaviorTimersSpec.scala]: I expect these last two are racy in whether they cover what they seem to cover, but I guess there isn't really any way around it, right? At least they aren't racy in a way that should fail now and then.

**Review by johanandren — APPROVED** [review]
LGTM

File: akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/EventSourcedBehaviorTimersSpec.scala (hunk 2 above)
Comment (patriknw) [reply akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/EventSourcedBehaviorTimersSpec.scala]: yes, that's why I wrote "likely", and the outcome should be the same

File: akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/EventSourcedBehaviorTimersSpec.scala (hunk 1 above)
Comment (patriknw) [reply akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/EventSourcedBehaviorTimersSpec.scala]: ok, I'll do that in https://github.com/akka/akka/pull/26579
although it's all over the place when we have the real event filter in place we should search for all usages of the old one

**Pull request merged**